lib/travis/api/app/endpoint/jobs.rb

```diff
@@ -80,8 +80,12 @@ class Jobs < Endpoint
         elsif (!resource || resource.archived?)
           # the way we use responders makes it hard to validate proper format
           # automatically here, so we need to check it explicitly
-          if accepts?('text/plain')
-            archived_log_path = archive_url("/jobs/#{params[:job_id]}/log.txt")
+          if accepts?('text/plain') || request.user_agent.to_s.start_with?('Travis')
```

Comment:
I think I should add that I arrived at the solution by getting the `travis logs {job_number}` command to work locally with this patch: https://gist.github.com/meatballhat/e56a9e5d0bd3e36fe9053aca207efa01